x/staking/keeper/validator.go

```diff
@@ -403,34 +403,26 @@ func (k Keeper) GetLastValidators(ctx context.Context) (validators []types.Valid
 // GetUnbondingValidators returns a slice of mature validator addresses that
 // complete their unbonding at a given time and height.
 func (k Keeper) GetUnbondingValidators(ctx context.Context, endTime time.Time, endHeight int64) ([]string, error) {
-	store := k.storeService.OpenKVStore(ctx)
-
-	bz, err := store.Get(types.GetValidatorQueueKey(endTime, endHeight))
+	timeBz := sdk.FormatTimeBytes(endTime)
+	timeBzL := len(timeBz)
```

Comment:
really unfortunate that we prefix time bytes with their length even when the size is constant and well-known. =(

Reply:
you can use here if you want k.ValidatorsQueue.KeyCodec().Size(endTime), instead of using sdk.FormatTimeBytes... At least this way we don't need to rely on format time bytes and we keep it in between collections. I would even comment this in the Keeper why we use three keys instead of using simply two.

Reply:
I don't think we can use `k.ValidatorQueue.KeyCodec().Size(endTime)` because key here should be of format `collections.Triple[uint64, time.Time, uint64]`, not just time(endTime). And it gives size of the combined length of fields in key.

Reply:
I mean, it's a good workaround, unfortunate that we had to do it but it works lol. Once we finish up with all these migrations we'll most definitely do a lot of refactor so we can do proper migrations then

Reply:
I see, then @likhita-809 maybe we can use sdk.TimeKey.Size directly, sorry for being pedantic about this, but I'd rather we remove every form of bytes conversion in state from production code, testing is fine to have it.

Reply:
Although, time key size is constant so in theory we could use the same number everywhere: https://github.com/cosmos/cosmos-sdk/blob/main/types/collections.go#L210